File: video-service/src/client/java/com/boclips/videos/service/client/spring/VideoServiceHealthIndicator.java
package com.boclips.videos.service.client.spring;

import com.boclips.videos.service.client.VideoServiceClient;
import org.springframework.boot.actuate.health.AbstractHealthIndicator;
import org.springframework.boot.actuate.health.Health;

public class VideoServiceHealthIndicator extends AbstractHealthIndicator {
    private final VideoServiceClient videoServiceClient;

    public VideoServiceHealthIndicator(VideoServiceClient videoServiceClient) {
        this.videoServiceClient = videoServiceClient;
    }


    @Override
    protected void doHealthCheck(Health.Builder builder) throws Exception {
        try {
            videoServiceClient.existsByContentPartnerInfo("non-existing-provider", "video-id");
            builder.up();
        } catch (Exception e) {
            builder.down();
        }
    }
}

File: video-service/src/client/kotlin/com/boclips/videos/service/client/spring/VideoServiceHealthIndicator.kt
package com.boclips.videos.service.client.spring

import com.boclips.videos.service.client.VideoServiceClient
import org.springframework.boot.actuate.health.AbstractHealthIndicator
import org.springframework.boot.actuate.health.Health

class VideoServiceHealthIndicator(val videoServiceClient: VideoServiceClient) : AbstractHealthIndicator() {
    override fun doHealthCheck(builder: Health.Builder) {
        try {
            videoServiceClient.existsByContentPartnerInfo("non-existing-provider", "video-id")
            builder.up()
        } catch (e: Exception) {
            builder.down()
        }
    }
}